wideq/ac.py

```diff
@@ -247,3 +654,87 @@ def fan_speed(self):
     def is_on(self):
         op = ACOp(self.lookup_enum('Operation'))
         return op != ACOp.OFF
+
+    @property
+    def wdirleftright_state(self):
+        return WDIRLEFTRIGHT(self.lookup_enum('WDirLeftRight'))
+
+    @property
+    def wdirupdown_state(self):
+        return ACETCMODE(self.lookup_enum('WDirUpDown'))    
+
+    @property
+    def airclean_state(self):
+        return AIRCLEAN(self.lookup_enum('AirClean'))
+
+    @property
+    def wdirvstep_state(self):
+        return WDIRVSTEP(self.data['WDirVStep'])
+
+    @property
+    def wdirhstep_state(self):
+        return WDIRHSTEP(self.data['WDirHStep'])
+
+    @property
+    def fourvain_wdirvstep_state(self):
+        return FOURVAIN_WDIRVSTEP(self.data['WDirVStep'])
+
+    @property
+    def sac_airclean_state(self):
+        return ACETCMODE(self.lookup_enum('AirClean'))    
+    
+    @property
+    def icevalley_state(self):
+        return ACETCMODE(self.lookup_enum('IceValley'))
+    
+    @property
+    def longpower_state(self):
+        return ACETCMODE(self.lookup_enum('FlowLongPower'))
+    
+    @property
+    def autodry_state(self):
+        return ACETCMODE(self.lookup_enum('AutoDry'))
+    
+    @property
+    def smartcare_state(self):
+        return ACETCMODE(self.lookup_enum('SmartCare'))
+    
+    @property
+    def sensormon_state(self):
+        return ACETCMODE(self.lookup_enum('SensorMon'))
+    
+    @property
+    def powersave_state(self):
+        return ACETCMODE(self.lookup_enum('PowerSave'))
+
+    @property
+    def jet_state(self):
+        return ACETCMODE(self.lookup_enum('Jet'))
+
+    @property
+    def humidity(self):
+        return self.data['SensorHumidity']
+    
+    @property
+    def sensorpm1(self):
+        return self.data['SensorPM1']
+    
+    @property
+    def sensorpm2(self):
+        return self.data['SensorPM2']
+    
+    @property
+    def sensorpm10(self):
+        return self.data['SensorPM10']
```

Comment:
What do these "sensorpmX" properties mean?

If possible, I would love to only expose things in this API that we actually understand. The LG service has a lot of inscrutable garbage in it that is probably not necessary for any client software, and we can keep things simpler by omitting that stuff.

Reply:
My guess is those values are PM2.5 and PM10 particulate matter, mostly as a smog.  
https://en.wikipedia.org/wiki/Particulates